keylime/src/tpm.rs

```diff
@@ -468,16 +465,11 @@ fn pubkey_to_tpm_digest<T: HasPublic>(
         }
     };
 
-    // SHA256
-    let mut hasher = openssl::sha::Sha256::new();
-    hasher.update(&keybytes);
-    let hashvec: Vec<u8> = hasher.finish().into();
-    keydigest.set(HashingAlgorithm::Sha256, Digest::try_from(hashvec)?);
-    // SHA1
-    let mut hasher = openssl::sha::Sha1::new();
-    hasher.update(&keybytes);
-    let hashvec: Vec<u8> = hasher.finish().into();
-    keydigest.set(HashingAlgorithm::Sha1, Digest::try_from(hashvec)?);
+    let hash_algo = HashingAlgorithm::from(algo);
+    let mut hasher = Hasher::new(hash_alg_to_message_digest(hash_algo)?)?;
+    hasher.update(&keybytes)?;
+    let hashvec = hasher.finish()?;
+    keydigest.set(hash_algo, Digest::try_from(hashvec.as_ref())?);
```

Comment:
Previously, it extended PCR16 with both SHA-1 and SHA-256. With this change, the slot seems to be only extended with the given single algorithm. Wouldn't that be a problem?

Reply:
Hi @ueno, thanks for your comment! 
I think it's safe here, let me explain more, please correct me if I'm wrong.
The working flow of Keylime attestation should be: `extend PCR16` --> `build PCR list` --> `get a quote with PCR list` --> `check the quote` --> `check PCR 16`.

1. The PCR list is built with [the specific hash algorithm](https://github.com/keylime/rust-keylime/blob/master/keylime/src/tpm.rs#L311).

2.  When the verifier checks the PCR 16, the expected value is generated with [the specific hash algorithm](https://github.com/keylime/keylime/blob/master/keylime/tpm/tpm_main.py#L203).

So, although it extended PCR16 with both SHA-1 and SHA-256 but looks like only the specific one will be measured.

Reply:
Yes, we check only the PCR 16 in the configured algorithm, so this change should not break anything.